src/frontend/src/optimizer/plan_node/logical_source.rs

```diff
@@ -343,8 +343,9 @@ impl ToStream for LogicalSource {
             }
             SourceNodeKind::CreateMViewOrBatch => {
                 // Create MV on source.
-                let use_shared_source = self.source_catalog().is_some_and(|c| c.info.is_shared())
-                    && self.ctx().session_ctx().config().rw_enable_shared_source();
+                // We only check rw_enable_shared_source is true when `CREATE SOURCE`.
+                // The value does not affect the behavior of `CREATE MATERIALIZED VIEW` here.
+                let use_shared_source = self.source_catalog().is_some_and(|c| c.info.is_shared());
```

Comment:
I changed this behavior because I found otherwise it's not easy to tell whether the SourceExecutor is shared. I guess this flexibility isn't very useful, so it's fine to be simpler.